app/src/main/java/com/nighthawkapps/wallet/android/ext/Dialogs.kt

```diff
@@ -84,17 +84,45 @@ fun Context.showScanFailure(
 
 fun Context.showCriticalProcessorError(error: Throwable?, onRetry: () -> Unit = {}): Dialog {
     return MaterialAlertDialogBuilder(this)
-        .setTitle("Processor Error")
-        .setMessage(error?.message ?: "Critical error while processing blocks!")
+        .setTitle(R.string.dialog_error_processor_critical_title)
+        .setMessage(error?.message ?: getString(R.string.dialog_error_processor_critical_message))
         .setCancelable(false)
-        .setPositiveButton("Retry") { d, _ ->
+        .setPositiveButton(R.string.dialog_error_processor_critical_button_positive) { d, _ ->
```

Comment:
previously, this button said "Retry" and would do nothing. Effectively, the user had two options:
- exit the app
- retry (which did nothing but close the dialog)

Now, when the user hits "Retry" the app will delete all data. That might be unexpected and may warrant better explanation to the user of what's about to happen. I would maybe consider two things here:

1. Remove this call to erase from this extension function
    because this helper function is letting the caller pass in the `onRetry` block. So the caller probably should have full control over what that means.
2. Migrate this erase behavior to the [call site in MainActivity](https://github.com/nighthawk-apps/nighthawk-wallet-android/blob/syncing-fixes-and-zip-313/app/src/main/java/com/nighthawkapps/wallet/android/ui/MainActivity.kt#L538)

----    

Now, the question becomes what should happen when the user hits "Retry"

### Easy route:
The easiest thing to do might be to give another dialog with a more thorough explanation of their options.    
- user taps 'retry'
- another dialog opens and gives a positive and negative option
  - the negative option just closes the dialog and does nothing
  - the positive option attempts a rescan

Perhaps,  the dialog says something like:
> Would you like to attempt to fix this issue by rescanning the blockchain?
> This might take over an hour to complete.
> [ No ] [ Rescan ]

### More work:
Instead of two dialogs, you could put a third button on the first dialog. This would probably require making a separate `Context.show` extension function inside `Dialogs.kt` and leveraging MaterialDialogBuilder's `setNeutralButton` and changing the dialog message to something that explains the 3 options with "wait" being the neutral one. Perhaps:

> Critical error while processing blocks!
> Would you like to wait it out? Exit the app? 
> Or attempt to fix this issue by rescanning the blockchain (which could take a while)?
> [ Exit ] [ Wait ] [ Rescan ]

Reply:
Done.